src/pages/create-organization.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { useState } from "react"
+import { useLocation } from "wouter"
+import { Helmet } from "react-helmet-async"
+import Header from "@/components/Header"
+import { Button } from "@/components/ui/button"
+import { Input } from "@/components/ui/input"
+import { Label } from "@/components/ui/label"
+import { useOrganizations } from "@/hooks/use-organizations"
+import { Check, X } from "lucide-react"
+import toast from "react-hot-toast"
+import { Organization } from "fake-snippets-api/lib/db/schema"
+
+interface FormErrors {
+  name?: string
+  displayName?: string
+  description?: string
+}
+
+export const CreateOrganizationPage = () => {
+  const [, setLocation] = useLocation()
+  const { createOrganization, checkNameAvailability } = useOrganizations()
+
+  const [formData, setFormData] = useState<
+    Pick<Organization, "github_handle" | "is_personal_org">
+  >({
+    github_handle: "",
+    is_personal_org: true,
+  })
+
+  const [errors, setErrors] = useState<FormErrors>({})
+  const [isLoading, setIsLoading] = useState(false)
+  const [nameAvailable, setNameAvailable] = useState<boolean | null>(null)
+  const [checkingName, setCheckingName] = useState(false)
+
+  const checkName = async (name: string) => {
+    if (!name || name.length < 3) {
+      setNameAvailable(null)
+      return
+    }
+
+    setCheckingName(true)
+
+    try {
+      const isAvailable = await checkNameAvailability(name)
+      setNameAvailable(isAvailable)
+    } catch (error) {
+      console.error("Error checking name availability:", error)
+      setNameAvailable(null)
```

Comment:
why are we handling this case... if it's likely to happen we should be using a react toast as well, but i don't understand why it would happen